catalog/ui/catalog-ui-search/src/main/webapp/component/input/input.view.js

```diff
@@ -52,6 +52,22 @@ define([
             this.handleEdit();
             this.handleReadOnly();
             this.handleValue();
+            this.handleValidation();
+        },
+        onAttach: function() {
+            this.listenForChange();
+        },
+        listenForChange: function(){
+            this.$el.on('change keyup input', function(){
+                this.model.set('value', this.getCurrentValue());
+                this.handleValidation();
+            }.bind(this));
+        },
+        handleValidation: function(){
+            this.$el.toggleClass('has-validation-issues', !this.isValid());
+        },
+        isValid: function(){
+            return true; //overwrite on a per input basis   
```

Comment:
The other inputs that people make overwrite this function.  By default, it returns true, but a special input (such as the input-enum) can do it's own validation and return true or false.